sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/MutationSizeEstimator.java

```diff
@@ -153,6 +161,21 @@ private static long estimateArrayValue(Value v) {
         return 12L * v.getDateArray().size();
       case TIMESTAMP:
         return 12L * v.getTimestampArray().size();
+      case NUMERIC:
+        totalLength = 0;
+        for (BigDecimal n : v.getNumericArray()) {
+          if (n == null) {
```

Comment:
Is size estimation all that is needed for adding support for this type ? Don't we have to handle the values that are read ?

Reply:
Yes:
The feature to support writing NUMERIC in BEAM was added in #12818  (BEAM-10875), but missed adding support for using java BigDecimal type support to the part of the code where the size of mutated values are calculated.
This PR fixes that.